example/service/s3/getObjectWithProgress/getObjectWithProgress.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"regexp"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"strings"
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+type progressWriter struct {
+	w  io.WriterAt
+	pb *pb.ProgressBar
+}
+
+func (pw *progressWriter) WriteAt(p []byte, off int64) (int, error) {
+	pw.pb.Add(len(p))
+	return pw.w.WriteAt(p, off)
+}
+
+func byteCountDecimal(b int64) string {
+	const unit = 1000
+	if b < unit {
+		return fmt.Sprintf("%d B", b)
+	}
+	div, exp := int64(unit), 0
+	for n := b / unit; n >= unit; n /= unit {
+		div *= unit
+		exp++
+	}
+	return fmt.Sprintf("%.1f %cB", float64(b)/float64(div), "kMGTPE"[exp])
```

Comment:
`"kMGTPE"[exp]` is an interesting way to capture the size suffix, I would not of thought to do a literal string index lookup.

Reply:
Yeah, I can't take credit for this, it was from a helper method in a library I was using 😄